### PR TITLE
PR: Correctly handle the option to run in a dedicated console without clearing variables

### LINKED ISSUE #19983: Execute in dedicated console without clearing variables
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Currently the "Execute in dedicated console" option currently clears the variables in the dedicated console regardless of the value "Remove all variables before execution" check box. As far as I can see it's been this way for at least 5 years.

Tested on latest master, 5.3.3, 5.1.5 and 5.2.x all affected.

I have a fix that I can commit. 

### What steps reproduce the problem?

1. Run a script in a dedicated console using the Run configuration per file" dialog with the "Execute in dedicated console" enabled
2. make sure that the "Remove all variables before execution" option is unchecked
3. After execution run the file again and notice that variables are reset and the your interactive plot windows are closed

### What is the expected output? What do you see instead?
If I don't check the "Remove all variables before execution" option, I expect them to not be removed.

## Versions

* Spyder version: 6.0.0.dev0 702f89d0b (conda)
* Python version: 3.10.6 64-bit
* Qt version: 5.15.6
* PyQt5 version: 5.15.7
* Operating System: Linux 5.15.0-46-generic

### Dependencies

```# Mandatory:
atomicwrites >=1.2.0          :  1.4.1 (OK)
chardet >=2.0.0               :  5.0.0 (OK)
cloudpickle >=0.5.0           :  2.2.0 (OK)
cookiecutter >=1.6.0          :  2.1.1 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.31.1;<8.0.0       :  7.33.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.17.0 (OK)
keyring >=17.0.0              :  23.9.3 (OK)
nbconvert >=4.0               :  7.2.3 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.3 (OK)
pygments >=2.0                :  2.13.0 (OK)
pylint >=2.5.0;<3.0           :  2.15.5 (OK)
pylint_venv >=2.1.1           :  None (NOK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0          :  1.5.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.2.2             :  0.2.2 (OK)
qtawesome >=1.2.1             :  1.2.1 (OK)
qtconsole >=5.3.2;<5.4.0      :  5.4.0.dev0 (OK)
qtpy >=2.1.0                  :  2.2.1 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  65.5.0 (OK)
sphinx >=0.6.6                :  5.3.0 (OK)
spyder_kernels >=2.3.3;<2.4.0 :  3.0.0.dev0 (OK)
textdistance >=4.2.0          :  4.5.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.9 (OK)
xdg >=0.26                    :  0.28 (OK)
zmq >=22.1.0                  :  24.0.1 (OK)

# Optional:
cython >=0.21                 :  0.29.32 (OK)
matplotlib >=3.0.0            :  3.6.2 (OK)
numpy >=1.7                   :  1.23.4 (OK)
pandas >=1.1.1                :  1.5.1 (OK)
scipy >=0.17.0                :  1.9.3 (OK)
sympy >=0.7.3                 :  1.11.1 (OK)
```

### PR DESCRIPTION
**Pull request by bcolsen**
## Description of Changes

Added the option to run in a dedicated console without clearing variables. I forgot to make a branch in #19984 so I'm replacing it with this one.

* [ ] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Added the option to run in a dedicated console without clearing variables because it seems like that should be an option and I like my interactive plot windows to stay in place and not pop up all the time

### Issue(s) Resolved

Fixes #19983 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: bcolsen

### CODE REVIEW

File: spyder/app/tests/test_mainwindow.py [hunk 1]
```diff
@@ -759,10 +761,23 @@ def test_dedicated_consoles(main_window, qtbot):
     text = control.toPlainText()
     assert ('runfile' in text) and not ('Python' in text or 'IPython' in text)
 
-    # --- Clean namespace after re-execution ---
+    # --- Clean retained after re-execution ---
```
Comment (dalthviz) [inline spyder/app/tests/test_mainwindow.py:764]: Maybe this comment could be clearer writting something like `Check namespace retention after re-execution`?

**Review by dalthviz — COMMENTED** [review]
Thanks @bcolsen for the help with this! Left a suggestion for the comment on the new case added to the tests when changing the `clear_namespace` config.

Other than that this LGTM :+1:

File: spyder/app/tests/test_mainwindow.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/app/tests/test_mainwindow.py:764]: I agree with @dalthviz's comment too. Otherwise this PR looks good to me.

**Review by ccordoba12 — APPROVED** [review]
Thanks @bcolsen!

Note: The failure in our tests is unrelated to this.

**Pull request merged**